sympy/core/tests/test_relational.py

```diff
@@ -661,3 +661,13 @@ def test_issue_10304():
     assert d.is_comparable is False  # if this fails, find a new d
     e = 1 + d*I
     assert simplify(Eq(e, 0)) is S.false
+
+
+def test_issue_8571():
+    raises(TypeError, lambda: x*S.true)
+    raises(TypeError, lambda: x + S.true)
+    raises(TypeError, lambda: S.true + x)
+    raises(TypeError, lambda: S(2)*S.false)
+    raises(TypeError, lambda: S(3)*S.true)
+    raises(TypeError, lambda: S.true*S(3.0))
+    assert S(1)*S.false == False
```

Comment:
Then it may need a fix elsewhere in the code. File another bug, CC me.

You can put a failing test here as an `xtest` and then a future PR can fix it: I don't recall the exact syntax for expected fail tests at the moment (`xtest` is some other language), but google can probably tell you...